enterprise_access/apps/bffs/tests/test_handlers.py

```diff
@@ -0,0 +1,125 @@
+            **arguments
+        )
+        self.assertEqual(expected_output, base_handler.context.errors[0])
+
+    def test_base_handler_add_error_serializer_is_valid(self):
+        base_handler = self.base_handler
+        malformed_output = {
+            "developer_message": "No enterprise uuid associated to the user mock-uuid",
+        }
+        with self.assertRaises(ValidationError):
+            base_handler.add_error(**malformed_output)
+
+    def test_base_handler_add_warning_serializer(self):
+        base_handler = self.base_handler
+        expected_output = {
+            "developer_message": "Heuristic Expiration",
+            "user_message": "The data received might be out-dated",
+        }
+        # Define kwargs for add_warning
+        arguments = {
+            **expected_output,
+            "status": 113  # Add an attribute that is not explicitly defined in the serializer to verify
+        }
+        base_handler.add_warning(
+            **arguments
+        )
+        self.assertEqual(expected_output, base_handler.context.warnings[0])
+
+    def test_base_handler_add_warning_serializer_is_valid(self):
+        base_handler = self.base_handler
+        malformed_output = {
+            "user_message": "The data received might be out-dated",
+        }
+        with self.assertRaises(ValidationError):
+            base_handler.add_error(**malformed_output)
+
+
+class TestBaseLearnerPortalHandler(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.factory = RequestFactory()
+        self.mock_user = UserFactory()
+        self.faker = Faker()
+
+        self.mock_enterprise_customer_uuid = self.faker.uuid4()
+        self.request = self.factory.get('sample/api/call')
+        self.request.query_params = {
+            'enterprise_customer_uuid': self.mock_enterprise_customer_uuid
+        }
+        self.request.user = self.mock_user
+        self.context = HandlerContext(self.request)
+        self.base_handler = BaseHandler(self.context)
+        self.base_learner_portal_handler = BaseLearnerPortalHandler(self.base_handler)
+
+    # TODO: Test pertaining to currently stubbed out functions deferred for future tickets
+
+
+class TestDashboardHandler(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.factory = RequestFactory()
+        self.mock_user = UserFactory()
+        self.faker = Faker()
+
+        self.mock_enterprise_customer_uuid = self.faker.uuid4()
+        self.request = self.factory.get('sample/api/call')
+        self.request.query_params = {
+            'enterprise_customer_uuid': self.mock_enterprise_customer_uuid
+        }
+        self.request.user = self.mock_user
+        self.context = HandlerContext(self.request)
+        self.base_handler = BaseHandler(self.context)
+        self.base_learner_portal_handler = BaseLearnerPortalHandler(self.base_handler)
+        self.dasboard_handler = DashboardHandler(self.base_learner_portal_handler)
```

Comment:
[curious] Is it necessary to instantiate each of these yourself? When you instantiate `DashboardHandler`, you're already instantiating `BaseLearnerPortalHandler` and `BaseHandler` since they subclass each other.

E.g. I believe you could just do:

```python
self.dashboard_handler = DashboardHandler(self.context)
```

...without needing `self.base_handler` and `self.base_learner_portal_handler`.

(similar comment for the `TestBaseLearnerPortalHandler` class).